fix `distBVH` integration

`distBVH` integration tests are failing with:
```
/opt/src/NimbleSM/src/contact/parallel/bvh_contact_manager.cc:134:18: error: 'build_snapshot_tree_top_down' was not declared in this scope; did you mean 'bvh::build_snapshot_tree_top_down'?
  134 |     auto  tree = build_snapshot_tree_top_down(_a.elements);
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                  bvh::build_snapshot_tree_top_down
```

https://github.com/sandialabs/distBVH/actions/runs/14084678416/job/39445818515?pr=26

Add `bvh::` namespace for this call and any other failing ones.

--- a/src/contact/parallel/bvh_contact_manager.cc
+++ b/src/contact/parallel/bvh_contact_manager.cc
@@ -134,7 +134,8 @@ struct NarrowphaseFunc
     auto  tree = bvh::build_snapshot_tree_top_down(_a.elements);
 
     std::size_t j = 0;
-    for (auto&& elb : _b.elements) {
+    Kokkos::parallel_for(_b.elements.extent(0), [=, &resa, &resb, &j](int i) {
+      auto elb = _b.elements(i);
       query_tree_local(tree, elb, [&_a, &_b, &elb, &resa, &resb, this, j](std::size_t _i) {
         const auto&       face = _a.elements[_i];
         const auto&       node = elb;
@@ -157,7 +158,7 @@ struct NarrowphaseFunc
         }
       });
       ++j;
-    }
+    });
 
     return {resa, resb};
   }

--- a/src/contact/parallel/bvh_contact_manager.cc
+++ b/src/contact/parallel/bvh_contact_manager.cc
@@ -131,7 +131,7 @@ struct NarrowphaseFunc
     res.b      = bvh::narrowphase_result(sizeof(NarrowphaseResult));
     auto& resa = static_cast<bvh::typed_narrowphase_result<NarrowphaseResult>&>(res.a);
     auto& resb = static_cast<bvh::typed_narrowphase_result<NarrowphaseResult>&>(res.b);
-    auto  tree = build_snapshot_tree_top_down(_a.elements);
+    auto  tree = bvh::build_snapshot_tree_top_down(_a.elements);
 
     std::size_t j = 0;
     for (auto&& elb : _b.elements) {

